Update async and defer learn JS section with explanation for blocking the page (#29393)
mdn/content: files/en-us/learn/javascript/first_steps/what_is_javascript/index.md

Let's look at the difference between these two.

Scripts loaded using the `async` attribute will download the script without blocking the page while the script is being fetched.

> However, once the download is complete, the script will execute, which blocks the page from rendering. Blocking the page refers to the event in which the rest of the content on the web page is prevented from being processed and displayed to the user until the script finishes executing.

Reviewer: ```suggestion
However, once the download is complete, the script will execute, which blocks the page from rendering. This means that the rest of the content on the web page is prevented from being processed and displayed to the user until the script finishes executing.
```
